conntrack/conntrack.go

```diff
@@ -22,11 +22,20 @@ import (
 	log "github.com/Sirupsen/logrus"
 )
 
-var conntrackDirections = []string{
+// For TCP/UDP, each conntrack entry holds two copies of the tuple
+// (src addr, dst addr, src port, dst port).  One copy for the original direction and one copy for
+// the reply direction.  This is how the kernel handles NAT: by looking up the tuple for a packet
+// by its original tuple and mapping onto the corresponding reply direction tuple (or vice versa).
+//
+// When we delete conntrack entries by IP address, we need to specify which element of the tuple
+// to look in.  This slice holds the flags corresponding to the fields we care about.  Since we're
+// deleting entries for local workload endpoints, either the endpoint originated the traffic, or it
+// received the traffic and replied to it.  In the originating case, the "original source" will be
+// set to the endpoint's IP; in the other case, the "reply source". Hence, it's sufficient to only
+// look in those two fields.
```

Comment:
To double-check my new understanding, then:

- If `orig-dst` is different from `reply-src`, that would only be because of a DNAT affecting connection _to_ the local workload, with `orig-dst` being the service or floating IP and `reply-src` being the endpoint IP.   Since we are deleting by endpoint IP, `reply-src` is the only useful lookup.

- If `reply-dst` is different from `orig-src`, that would only be because of an SNAT affecting outgoing connections _from_ the local workload, with `orig-src` being the endpoint IP - and again, because we are deleting by endpoint IP, `orig-src` is the only useful lookup.

That all seems watertight to me, so I'll approve now.